### DISCUSSION TITLE
click clone button with MultipleFileUploadEditor in detail

### DISCUSSION THREAD

**Discussion by wow-prime** [serenity-is/Serenity#6428] [Help] [answered]
When cloning a MasterDetailRelation entity that detail have attachment, this error occurs.
"For security reasons, only temporary files can be used in uploads!"

I try move file to temporary manually in "protected override void BeforeSave()"
![image](https://user-images.githubusercontent.com/19904096/175874445-3192ab2c-b8de-4193-b26a-efecaf10d0c2.png)

but result was wrong
![image](https://user-images.githubusercontent.com/19904096/175874722-74951fdf-3a3f-43ee-9e33-c05d184467f0.png)

Who knows how to solve this problem?

**Comment by edwardch** [comment, marked as answer]
I think you might need to set the file metadata first.

Check what `UploadStorage.SetOriginalName(temporaryFile, file.FileName);` does.

Here is some old code I used to manually add a file to a table:
```cs
var extension = Path.GetExtension(file);
var fileName = Path.GetFileName(file);

// Serenity file handler uses this relative path to the app's upload path setting
var relativeTemporaryPath = Path.Combine($"temporary/{Guid.NewGuid()}{extension}");

// The FileUpload in Serenity needs a metadata file
// containing the Original Name of the file
var metadata = new Dictionary<string, string>()
{
    [FileMetadataKeys.OriginalName] = fileName
};
UploadStorage.SetFileMetadata(relativeTemporaryPath, metadata, overwriteAll: false);

fileList.Add(new FileRow()
{
    Name = fileName,
    Path = relativeTemporaryPath
});
```

HTH

**Comment by wow-prime** [comment]
It's OK. Thanks.
